### PR TITLE
guard-proofreader-from-missing-translation-error

### PR DESCRIPTION
**Pull request by anathomical**
## WHAT
Add a guard clause to protect against missing activity on props
## WHY
Because Proofreader doesn't actually set this value
## HOW
Check if `activity` exists before trying to destructure it, and bail early if it isn't there

### Screenshots
image

### Notion Card Links
https://www.notion.so/quill/Student-sees-a-blank-page-when-they-click-the-Resume-button-for-the-activity-The-Supremes-106e439473f643a3bdafd3c010fe6304

### What have you done to QA this feature?
- Complete a Proofreader activity on staging
- Click "Save and exit"
- Confirm that when Resuming the activity, the page crashes when trying to load the follow-up Grammar activity
- Fix code and test
- Confirm by copying `data` blob from a known erroring production `ActiveActivitySession` to staging and confirm that Resuming does not crash the follow-up activity

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | No, there's no test coverage of the code with the edge case
Have you deployed to Staging? | Yes
Self-Review: Have you done an initial self-review of the code below on Github? | Yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Grammar/actions/session.ts [hunk 1]
```diff
@@ -132,7 +132,16 @@ const activityUID = (): string => { return getParameterByName('uid', window.loca
 const handleGrammarSession = (session) => {
   return dispatch => {
     if (session && Object.keys(session).length > 1 && !session.error) {
-      QuestionApi.getAllForActivity(activityUID()).then((allQuestions) => {
+
+      // activityUID will not be set in cases where a user is resuming a
```
Comment (samanthamjohn) [inline services/QuillLMS/client/app/bundles/Grammar/actions/session.ts:136]: Question: did we confirm that not having questions populated was causing our error? i.e. could we return an empty object, or maybe just return one question? It seems that proofreader activities do not use questions at all, so it seems wasteful to pull all the questions here.

File: services/QuillLMS/client/app/bundles/Grammar/components/grammarActivities/intro.tsx [hunk 2]
```diff
@@ -45,6 +45,10 @@ export default class Intro extends React.Component<IntroProps, IntroState> {
 
   translatedText = () => {
     const { activity, language } = this.props;
+
+    // Proofreader does not actually set the activity on props, so bail
+    if (!activity) return false
```
Comment (samanthamjohn) [inline services/QuillLMS/client/app/bundles/Grammar/components/grammarActivities/intro.tsx:50]: looks good!

**Review by samanthamjohn — CHANGES_REQUESTED** [review]
I think the fetchQuestions could be done in another PR, because it's related to a different potential bug.

**Review by happythenewsad — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Grammar/actions/session.ts (hunk 1 above)
Comment (anathomical) [reply services/QuillLMS/client/app/bundles/Grammar/actions/session.ts:136]: What's happening is that if you don't get all the questions right on the Proofreader activity, the system attempts to start the follow-up Grammar activity.  But the uid is still not set in the URL, so the Grammar activity is what's crashing.

**Review by samanthamjohn — APPROVED** [review]
